quiz/src/Qsnblock.tsx

```diff
@@ -0,0 +1,88 @@
+import { useEffect, useState } from "react";
+
+import { useReduce } from "./Reducer-context";
+import { quiz } from "./Data/quiz.types";
+
+import RotateLeft from "@material-ui/icons/RotateLeft"
+import CircularProgressWithLabel from "./CircularProgressWithLabel";
+import { Button, Box, Grid } from "@material-ui/core";
+
+export function Qsnblock({ typedata }: { typedata: quiz }) {
+  let { currentQsnNo, dispatch} = useReduce();
+  const [time, settime] = useState(15);
+  const totaltime=15
+  useEffect(() => {
+    let timer = setInterval(() => {
+      settime((prev) => prev - 1);
+    }, 1000);
+
+    if (time === 0) {
+      dispatch({ type: "SKIP" });
+      settime(15);
+    }
+    return () => {
+      clearInterval(timer);
+    };
+  }, [dispatch, time]);
+
+  let { question, options, points } = typedata.questions[currentQsnNo - 1];
+
+  return (
+    <>
+      <Box textAlign="center" style={{width:"100%"}}>
+        <div>{question}</div>
+        <br/>
+        <CircularProgressWithLabel value={(time/totaltime)*100} total={totaltime} />
+        <h6>Number of point- {points}</h6>
+        <Grid container spacing={3} justify="center" style={{width:"100%"}}>
+          {options.map((item, index) => {
+            return (
+              
+                <Grid item xs={10} sm={5} md={4} lg={3}>
+                  <Button
+                    variant="outlined"
+                    style={{width:"100%",height:"100%"}}
+                    color="primary"
+                    onClick={() => {
+                      if (item.isRight === false) {
+                        dispatch({
+                          type: "DECREMENT_SCORE",
+                          payload: { score: points }
+                        });
+                        settime(15);
+                      } else {
+                        dispatch({
+                          type: "INCREMENT_SCORE",
+                          payload: { score: points }
+                        });
+                        settime(15);
+                      }
+                    }}
```

Comment:
Extract this onClick function into a different function and just use the name of the function here.